### PR TITLE
feat: add support for Radiomaster MT12

### PR DESCRIPTION
**Pull request by raphaelcoeffic**
Adds support for Radiomaster MT 12 surface radio

https://youtu.be/YCcXS6_FBbc?si=GId-BOt7E-CFwsTZ

COMPANION SUPPORT NOT INCLUDED YET

See below for explanation of RGB LED control  : https://github.com/EdgeTX/edgetx/pull/4277#issuecomment-1791993675

Servo test port functionality is not yet implemented.

### CODE REVIEW

File: radio/src/lua/api_general.cpp [hunk 1]
```diff
@@ -3202,7 +3202,7 @@ LROT_BEGIN(etxcst, NULL, 0)
   LROT_NUMENTRY( EVT_VIRTUAL_ENTER_LONG, EVT_KEY_LONG(KEY_ENTER) )
   LROT_NUMENTRY( EVT_VIRTUAL_EXIT, EVT_KEY_BREAK(KEY_EXIT) )
 #elif defined(NAVIGATION_X7) || defined(NAVIGATION_X9D)
-#if defined(RADIO_TX12) || defined(RADIO_TX12MK2) || defined(RADIO_BOXER) || defined(RADIO_ZORRO) || defined(RADIO_POCKET) || defined(RADIO_T8) || defined(RADIO_COMMANDO8)
+#if defined(RADIO_TX12) || defined(RADIO_TX12MK2) || defined(RADIO_BOXER) || defined(RADIO_ZORRO)  || defined(RADIO_T8) || defined(RADIO_COMMANDO8) || defined(RADIO_MT12)
```
Comment (philmoz) [inline radio/src/lua/api_general.cpp:3205]: Did you mean to remove RADIO_POCKET?

Comment (3djc) [reply radio/src/lua/api_general.cpp:3205]: Nop, rebase issue. Fixing

**Comment by 3djc** [comment]
RGB leds are driven by LUA scripts.
A set of basic scripts is included here: https://github.com/EdgeTX/edgetx-sdcard/tree/master/sdcard/bw128x64/SCRIPTS/RGBLED

Currently, after setting up a SF to activate the RGB led, you might have to reboot for it to take effect, we hope to overcome that limitation shortly.

Usage exemple:
image

image

This allows changing the color of the led based on throttle position (moving forward/backward)

File: radio/src/targets/taranis/hal.h [hunk 2]
```diff
@@ -890,6 +935,15 @@
...
   #define SWITCHES_GPIO_PIN_F           LL_GPIO_PIN_5 // PB.05
+#elif defined(RADIO_MT12)
+  // ADC based switch/pot
+#elif defined(RADIO_TPRO)
+  // SW2
+  #define FUNCTION_SWITCH_2             SF
+  #define STORAGE_SWITCH_F
+  #define HARDWARE_SWITCH_F
+  #define SWITCHES_GPIO_REG_F           GPIOG
+  #define SWITCHES_GPIO_PIN_F           LL_GPIO_PIN_6  // PG.06
```
Comment (pfeerick) [inline radio/src/targets/taranis/hal.h:946]: Was this something that got missed before with the TPRO somehow? I only ask as it seems out of place :shrug:

Comment (pfeerick) [inline radio/src/targets/taranis/hal.h:946]: Was this something that got missed before with the TPRO somehow? I only ask as it seems out of place 🤷

Comment (3djc) [reply radio/src/targets/taranis/hal.h]: indeed

**Review by pfeerick — APPROVED** [review]
LGTM with internal ELRS MT12, also tested with external MPM

**Pull request merged**